lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -217,6 +219,25 @@ def compile_props(constant)
           end.join("\n")
         end
 
+        sig { params(constant: Module).returns(String) }
+        def compile_enums(constant)
+          return "" unless T::Enum > constant
+
+          # T::Enum doesn't currently give a nice way to get the actual enum name so you
+          # need to parse it from the string representation which is in the form `#<SomeEnum::A>`
+          enums = T.cast(constant, T::Enum).values.map do |value|
+            value.to_s.match(/#<([\w:]+)>/)[1].split('::').last
```

Comment:
I think we can do:
```suggestion
            value.instance_variable_get(:@const_name).to_s
```
as well.

Reply:
I was originally thinking that but wasn't sure how you'd feel about reaching into a private instance method. I think it's much better than the regex tbh 👍

Reply:
Yeah, I'd rather that we explicitly reach into the functionality instead of relying on a name serialization side-effect to grab this. It would make it easier to troubleshoot if the functionality changes down the line.